### PR TITLE
docs(cdk/coercion): expand overview

### PR DESCRIPTION
**Pull request by crisbeto**
Companion PR to https://github.com/angular/material.angular.io/pull/994. Expands the overview of the coercion package so that we have more than one sentence to show.

### CODE REVIEW

File: src/cdk/coercion/coercion.md [hunk 1]
```diff
@@ -1,3 +1,59 @@
...
+
+  // `coerceElement` allows you to accept either an `ElementRef`
+  // or a raw DOM node and to always return a DOM node.
+  getElement(elementOrRef: ElementRef<HTMLElement> | HTMLElement): HTMLElement {
+    return coerceElement(elementOrRef);
+  }
+
+  // Required so that the template type checker can infer the type of the coerced inputs.
+  static ngAcceptInputType_disabled: BooleanInput;
+  static ngAcceptInputType_greetDelay: BooleanInput;
```
Comment (andrewseguin) [inline src/cdk/coercion/coercion.md:57]: `NumberInput`

File: src/cdk/coercion/coercion.md [hunk 2]
```diff
@@ -1,3 +1,59 @@
...
+  @Input()
+  get disabled() { return this._disabled; }
+  set disabled(value: any) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+  private _disabled = false;
+
+  // Using `coerceNumberProperty` allows for any value to be set to `greetDelay` while ensuring
+  // that a number is always returned. E.g. the consumer can write:
+  // `<my-button greetDelay="500"></my-button>` instead of `<my-button [greetDelay]="500"></my-button>`.
```
Comment (andrewseguin) [inline src/cdk/coercion/coercion.md:34]: I wonder if most users will see that `greetDelay="500"` means its passing a string, and that's issue the coercion is solving. Could we add a sentence explaining that its turning the string into a number?

**Review by andrewseguin — APPROVED** [review]
Good start to the doc! I wonder if we eventually want to add some kind of table that helps users know what different values turn into, e.g. `` turns to `true`, `"false"` is `false`, etc

File: src/cdk/coercion/coercion.md [hunk 3]
```diff
@@ -1,3 +1,59 @@
-### Coercion
-
 Utility functions for coercing `@Input`s into specific types.
+
+### Example
+
+```ts
```
Comment (jelbourn) [inline src/cdk/coercion/coercion.md:5]: Should we make this a full example? It's pretty extensive for a code block

File: src/cdk/coercion/coercion.md [hunk 4]
```diff
@@ -1,3 +1,59 @@
...
+  // The second parameter specifies a fallback value to be used if the value can't be
+  // parsed to a number.
+  @Input()
+  get greetDelay() { return this._greetDelay; }
+  set greetDelay(value: any) {
+    this._greetDelay = coerceNumberProperty(value, 0);
+  }
+  private _greetDelay = 0;
+
+  @HostListener('click')
```
Comment (jelbourn) [inline src/cdk/coercion/coercion.md:44]: Can you use `host:` here instead of `HostListener`?

File: src/cdk/coercion/coercion.md (hunk 3 above)
Comment (crisbeto) [reply src/cdk/coercion/coercion.md:5]: My reasoning for keeping it inline is that the example won't be particularly visual since the coercion only affects the public API of the component.

File: src/cdk/coercion/coercion.md [hunk 5]
```diff
@@ -1,3 +1,59 @@
...
+
+```ts
+import {Directive, ElementRef, HostListener} from '@angular/core';
+import {
+  coerceBooleanProperty,
+  BooleanInput,
+  NumberInput,
+  coerceNumberProperty,
+  coerceElement,
+  } from '@angular/cdk/coercion';
```
Comment (dev054) [inline src/cdk/coercion/coercion.md:13]: It seems like you have an unnecessary whitespace here:

```suggestion
} from '@angular/cdk/coercion';
```

File: src/cdk/coercion/coercion.md [hunk 6]
```diff
@@ -1,3 +1,59 @@
...
+    '(click)': 'greet()',
+  }
+})
+class MyButton {
+  // Using `coerceBooleanProperty` allows for the disabled value of a button to be set as
+  // `<my-button disabled></my-button>` instead of `<my-button [disabled]="true"></my-button>`.
+  // It also allows for a string to be passed like `<my-button disabled="true"></my-button>`.
+  @Input()
+  get disabled() { return this._disabled; }
+  set disabled(value: any) {
```
Comment (dev054) [inline src/cdk/coercion/coercion.md:28]: Sorry to ask, but is `any` correct here (and in the other example)? I ask therefore, because as it is a bad practice, IMHO it shouldn't be in official docs as this can influence bad habits for other devs. Note that a simple boolean would solve it (as you already does in majority of your components).
